shotover-proxy/src/frame/cassandra.rs

```diff
@@ -385,27 +386,43 @@ impl CassandraFrame {
         cursor.write_all(&[self.operation.to_opcode().into()]).ok();
 
         serialize_with_length_prefix(&mut cursor, |cursor| {
-            if let Tracing::Response(Some(uuid)) = self.tracing {
-                cursor.write_all(uuid.as_bytes()).ok();
-            }
+            if Compression::None == compression || self.operation.to_opcode() == Opcode::Startup {
```

Comment:
For this bit `|| self.operation.to_opcode() == Opcode::Startup`, I'm not sure where this help, say the client accidentally compressed the startup message, trying to interpret it as non-compressed isnt going to save us.